### ISSUE TITLE
Configuration: Cannot process argument transformation on parameter 'ResourceModuleTuplesToImport'. Cannot convert the "System.Collections.ArrayList" value of type "System.Collections.ArrayList" to type "System.Tuple`3[System.String[],Microsoft.PowerShell.Commands.ModuleSpecification[],System.Version]".

### ISSUE THREAD

**Issue by laxmip01** [microsoft/Microsoft365DSC#2054]
Configuration: Cannot process argument transformation on parameter 'ResourceModuleTuplesToImport'. Cannot convert the "System.Collections.ArrayList" value of type "System.Collections.ArrayList" to type "System.Tuple`3[System.String[],Microsoft.PowerShell.Commands.ModuleSpecification[],System.Version]".

When attempting to execute the extracted configuration.ps1 file it throws above error.

**Comment by andikrueger** [comment]
Any chance, you could narrow it down to an example configuration, so that we get a chance to work on this issue? The error message does not show, what kind of resources is causing this error with what kind of settings.

**Comment by ykuijs** [comment]
This looks like an issue with the way the Import-DscResource cmdlet is being called. 

As Andy suggests: Please share the configuration you are trying to apply, so we can check what you are trying to do and assist in resolving this issue. Of course, you can obfuscate any sensitive information before sharing it here.

**Comment by thacold1** [comment]
I receive this error also when trying to compile my MOF file. 

PS C:\JetskiNation\DSC\Teams> .\TeamsConfigsP1.ps1
Configuration: Cannot process argument transformation on parameter 'ResourceModuleTuplesToImport'. Cannot convert the "System.Collections.ArrayList" value of type "System.Collections.ArrayList" to type "System.Tuple`3[System.String[],Microsoft.PowerShell.Commands.ModuleSpecification[],System.Version]".

**Comment by andikrueger** [comment]
Could you please share the top
part of your configuration. There might be something wrong with the import of the dsc resource.

**Comment by laxmip01** [comment]
Please let me know what is the configuration you are looking for? I tried exporting the SPO tenant settings export DSC and it created a .ps1 file of M365configuration. I'm trying to compile that .PS1 to MOF thats when this error is triggered.

**Comment by ykuijs** [comment]
@laxmip01 Exactly, please share that generated PS1 file

**Comment by andikrueger** [comment]
Thanks for the upper part. This looks fine. Could you share the whole file. Please make sure to remove critical data, that should not be exposed.

**Comment by laxmip01** [comment]
[M365TenantConfig.txt](https://github.com/microsoft/Microsoft365DSC/files/9063493/M365TenantConfig.txt)

**Comment by andikrueger** [comment]
Please make sure to run M365DSC within PowerShell Version 5.1. I did some quick testing with this configuration and I can see this error in PS 7.2.4 but not in PS 5.1.

**Comment by laxmip01** [comment]
Can i use Powershell version 5.1 on a Windows 10 machine

**Comment by laxmip01** [comment]
I tried it in Powershell 5.1 and was able to compile the MOF File using the .ps1 and also managed to run Start-DSCConfiguration -Path C:\DemoM365DSC\M365TenantConfig -Wait -Verbose -Force this modified the settings what i changed from .ps1 file.

**Comment by ykuijs** [comment]
We are testing PS7, but there are some challenges:
- Some of the modules we depend on require Windows PowerShell 5.1. There are some workarounds, which we have tested successfully in PS7.1, but that version is no longer supported (the even minor versions like 7.0 and 7.2 are the long term supported versions, https://docs.microsoft.com/en-us/powershell/scripting/install/installing-powershell-on-windows?view=powershell-7.2#supported-versions-of-windows).
- The PSDesiredStateConfiguration module has been removed from PowerShell 7.2. DSCv3 is currently in preview and is a major overhaul in comparison with earlier versions. More information: https://docs.microsoft.com/en-us/powershell/scripting/dsc/overview?view=powershell-7.2

In summary: We are working on fully supporting PS7, but are relying on the product groups to support that as well.

**Comment by laxmip01** [comment]
Thanks guys for your support didn't expect finally PS 5.1 works :) but I guess for the Azure Pipeline we must adopt PS7 or even pS5 supported
